pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -96,6 +96,8 @@ func (l *ScrapeConfigList) DeepCopyObject() runtime.Object {
 // ScrapeConfigSpec is a specification of the desired configuration for a scrape configuration.
 // +k8s:openapi-gen=true
 type ScrapeConfigSpec struct {
+	// The job name assigned to scraped metrics by default.
+	JobName string `json:"jobName,omitempty"`
```

Comment:
```suggestion
	//+optional
	JobName *string `json:"jobName,omitempty"`
```

We want to follow K8s API conventions for optional fields
(adding optional markers and using pointers)

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

Reply:
added, thanks

Reply:
@janluak can you apply Arthur's comment? Thanks!

Reply:
Sorry, I am a bit confused: the code already contains the `//+optional` comment:
<img width="858" alt="image" src="https://github.com/prometheus-operator/prometheus-operator/assets/12571346/26ab3fcc-d8a8-4e0a-a285-1c157e541b6a">
What did I miss? (see code also [here](https://github.com/janluak/prometheus-operator/blob/feature/crd-scrapeConfig-with-jobName/pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go))

I committed the change myself tough to have the commit message correct: I didn't know how to stay in the structure if using the GitHub-UI to do the commit...